### ISSUE TITLE
customize folder to save files

### ISSUE THREAD

**Issue by ToothlessR** [aleksey-rezvov/obsidian-local-images#9]
Nice plugin! Thanks.
It would be nice if user can set up the folder to save the files. For example, in media/%mdfilename%/, so all the saved files would be stuck unorderly in one folder.

**Comment by meskill** [comment]
Another option for customization - reuse obsidian `default location for new attachments`

As I am using a special subfolder for attachments and this behavior will be convenient for me

**Comment by aleksey-rezvov** [comment]
> Nice plugin! Thanks. It would be nice if user can set up the folder to save the files. For example, in media/%mdfilename%/, so all the saved files would be stuck unorderly in one folder.

Thank you! 

It should not be hard, I will try to do it when I will be free.



> Another option for customization - reuse obsidian `default location for new attachments`

I didn't know about such an option, thank you!

**Comment by j-peeters** [comment]
First, thank you for your very helpful plugin. 

I came here to ask something similar to the previous posters. I would love to store my local images in the 'assets' subfolder of the folder the note resides in. 

I tried using '/assets' in the Media folder field but with no luck. It kept storing the local files in the assets folder in the root of my vault.

So. I'm curious if this is already possible and if I'm missing something. Thanks

**Comment by snowshoes** [comment]
> Another option for customization - reuse obsidian `default location for new attachments`
> 
> As I am using a special subfolder for attachments and this behavior will be convenient for me

I've applied this option but it doesn't work, all images remain downloading to default vault directory instead of the path I configured in `Settings -> Files & Links -> default location for new attachments`

**Comment by meskill** [comment]
@snowshoes That's the point of the issue - currently plugin does not support changing folder for the attachments. We are waiting for the developer to implement this

**Comment by zeltak** [comment]
thx so much for this super useful @aleksey-rezvov!

im also wondering how one can configure the local folder location

I use a `assest` folder under each folder is 

```
./assets
```

I tried doing that in the plugin:

![image](https://user-images.githubusercontent.com/498410/153593979-8aea2bfd-7800-49e3-88a0-b37e61286075.png)


with no success..is that possible?

thx!

Z

**Comment by j-peeters** [comment]
@zeltak currently not. We are all waiting for a reply from the author on this.

**Comment by i-iooi-i** [comment]
> 

I found it ready to use. Just fill in the folder name manually.

**Comment by j-peeters** [comment]
> > 
> 
> I found it ready to use. Just fill in the folder name manually.

Can you confirm that you are being able to have the extension store files in a manually set subfolder of the folder in which the note resides. For example:
`./assets`

**Comment by i-iooi-i** [comment]
![image](https://user-images.githubusercontent.com/1827701/161706837-76f3e1f6-425e-4ac3-946c-53e30f5d6589.png)

Like this, I just fill in the folder name and I can use it successfully. ". /" was not added.

**Comment by j-peeters** [comment]
Yes that 

> ![image](https://user-images.githubusercontent.com/1827701/161706837-76f3e1f6-425e-4ac3-946c-53e30f5d6589.png)
> 
> Like this, I just fill in the folder name and I can use it successfully. ". /" was not added.

Yes, that might work but what people are trying to achieve is to have their local files stored in the subfolder of the folder in which the note resides. That is something different from the setup you have. All your assets are stored in one folder in your case as far as I can tell.

The scenario with the subfolder would be useful when you have daily notes in weekly folders and want the assets of that week collect in a subfolder of your weekly.

**Comment by i-iooi-i** [comment]
> Yes that
> 
> > ![image](https://user-images.githubusercontent.com/1827701/161706837-76f3e1f6-425e-4ac3-946c-53e30f5d6589.png)
> > Like this, I just fill in the folder name and I can use it successfully. ". /" was not added.
> 
> Yes, that might work but what people are trying to achieve is to have their local files stored in the subfolder of the folder in which the note resides. That is something different from the setup you have. All your assets are stored in one folder in your case as far as I can tell.
> 
> The scenario with the subfolder would be useful when you have daily notes in weekly folders and want the assets of that week collect in a subfolder of your weekly.

Yes, I use this method to uniformly store images, and then use the observer gallery（https://github.com/Darakah/obsidian-gallery） to uniformly view and manage images.

**Comment by j-peeters** [comment]
Nice plugin, thanks for the share. 
I still hope to be able to store my local-ed files in subfolders.

**Comment by lpreterite** [comment]
> thx so much for this super useful @aleksey-rezvov!
> 
> im also wondering how one can configure the local folder location
> 
> I use a `assest` folder under each folder is
> 
> ```
> ./assets
> ```
> 
> I tried doing that in the plugin:
> 
> ![image](https://user-images.githubusercontent.com/498410/153593979-8aea2bfd-7800-49e3-88a0-b37e61286075.png)
> 
> with no success..is that possible?
> 
> thx!
> 
> Z

I made a PR to support setting media files to be saved to the `assets` directory under each directory. See [PR#47](https://github.com/aleksey-rezvov/obsidian-local-images/pull/47) for details.
